libs/hdf-converters/src/converters-from-hdf/splunk/reverse-splunk-mapper.ts

```diff
@@ -432,48 +431,64 @@ export class FromHDFToSplunkMapper extends FromAnyBaseConverter {
+      `Logging into Splunk instance at ${hostname} with user ${config.username}`
     );
-    logger.verbose('Got Execution: ' + filename);
+
+    logger.verbose(`Found designated file to transfer: ${filename}`);
     const guid = createGUID(30);
-    logger.verbose('Using GUID: ' + guid);
-    return new Promise((resolve) => {
-      if (config.username && config.password) {
-        service.login((error: any) => {
-          this.handleSplunkError(error);
-        });
+    logger.verbose(`Using GUID: ${guid}`);
+
+    // Attempt to authenticate using given credentials
+    const username = (config.username ??= '');
+    const password = (config.password ??= '');
+    const authResponse = await this.axiosInstance.post(
+      `${hostname}/services/auth/login`,
+      new URLSearchParams({
+        username: username,
+        password: password
+      })
+    );
+    this.axiosInstance.defaults.headers.common[
+      'Authorization'
+    ] = `Bearer ${authResponse.data.sessionKey}`;
+
+    // Request all available indexes
+    const indexResponse = await this.axiosInstance.get(
+      `${hostname}/services/data/indexes`,
+      {
+        params: {count: 0}
       }
-      service.indexes().fetch(async (error: any, indexes: any) => {
-        if (error) {
-          this.handleSplunkError(error);
-        } else if (!indexes) {
+    );
+
+    return new Promise((resolve) => {
+      try {
+        // Report provided indexes
+        const indexes = indexResponse.data.entry;
+        if (indexes.length <= 0) {
           throw new Error(
-            'Unable to get available indexes, double-check your scheme configuration and try again'
+            'Unable to retrieve available indexes, double-check your scheme configuration and try again'
           );
         } else {
-          const availableIndexes: string[] = indexes
-            .list()
-            .map((index: {name: string}) => index.name);
-          logger.verbose(
-            `Available Indexes:  + ${availableIndexes.join(', ')}`
+          const indexNames: string[] = indexes.map(
+            (index: {name: string}) => index.name
           );
-          if (availableIndexes.includes(config.index)) {
-            const targetIndex = indexes.item(config.index);
-            logger?.verbose(`Have index ${targetIndex.name}`);
+          logger.verbose(`Available indexes: ${indexNames.join(', ')}`);
+
+          // Parse available indexes for user desired index
+          if (indexNames.includes(config.index)) {
+            const targetIndex = indexes.filter(
+              (index: {name: string}) => index.name == config.index
+            )[0];
+            logger.verbose(`Found index: ${targetIndex.name}`);
+
+            // Post given file(s) to identified index
             const splunkData = this.createSplunkData(guid, filename);
-            this.uploadSplunkData(targetIndex, splunkData)
+            this.uploadSplunkData(config, targetIndex, splunkData)
```

Comment:
there's no guarantee that this promise resolves by the end of the function.  clearer to deal with this issue using async/await.

Reply:
Changed as requested.